matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/room/peeking/PeekRoomTask.kt

```diff
@@ -65,23 +66,29 @@ internal class DefaultPeekRoomTask @Inject constructor(
         }
 
         // Is it a public room?
-        val publicRepoResult = when (getRoomDirectoryVisibilityTask.execute(GetRoomDirectoryVisibilityTask.Params(roomId))) {
-            RoomDirectoryVisibility.PRIVATE -> {
-                // We cannot resolve this room :/
-                null
-            }
-            RoomDirectoryVisibility.PUBLIC  -> {
+        val visibilityRes = tryOrNull("## PEEK: failed to get visibility") {
```

Comment:
I'm not a big fan of using `tryOrNull {}` around tasks which make network request. If there is no network, there is no need to go further.

Reply:
We don't have good mechanism to be sure we can reach the server, so how do you want to handle this?

Reply:
No network: special Exception. Getting 404 is something else for instance